Source/Basetypes/BigFloat.cs

```diff
@@ -210,7 +260,7 @@ private BIM maxsignificand()
     [Pure]
     public override string/*!*/ ToString() {
       Contract.Ensures(Contract.Result<string>() != null);
-      return value=="" ? String.Format("{0}x2^{1}", significand.ToString(), Exponent.ToString()) : value;
+      return value == "" ? String.Format("{0}{1}e{2}f{3}e{4}", isNeg ? "-" : "", significand.ToString(), exponent.ToString(), significandSize.ToString(), exponentSize.ToString()) : String.Format("0{0}{1}e{2}", value, significandSize.ToString(), exponentSize.ToString());
```

Comment:
Please break this into several lines. Maybe also think about splitting it into several statements.